src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -337,25 +342,46 @@
                 });
     }
 
+    public Uni<List<Endpoint>> loadProperties(List<Endpoint> endpoints) {
+        // Group endpoints in types and load in batches each type.
+        // Currently only webhook types have properties.
+        endpoints = endpoints.stream().distinct().collect(Collectors.toList());
```

Comment:
`Endpoint` has `equals` and `hashCode` implementations. Creating a `Set` from the `List` would probably be faster than the dealing with the duplicates with a stream.